src/src/com/microsoft/aad/adal/StorageHelper.java

```diff
@@ -123,124 +129,94 @@
+    private KeyPair mKeyPair;
     private String mBlobVersion;
-
-    private SecretKey mKey = null, mMacKey = null;
-
-    private static SecretKey sSecretKeyFromAndroidKeyStore = null;
+    private SecretKey mKey = null;
+    private SecretKey mMacKey = null;
+    private SecretKey mSecretKeyFromAndroidKeyStore = null;
 
     public StorageHelper(Context ctx) {
         mContext = ctx;
         mRandom = new SecureRandom();
     }
 
     /**
-     * Get Secret Key based on API level to use in encryption. Decryption key
-     * depends on version# since user can migrate to new Android.OS
+     * encrypt text with current key based on API level
      *
-     * @return
-     * @throws NoSuchAlgorithmException
+     * @param clearText Clar text to encrypt. 
+     * @return Encrypted blob.
+     * @throws GeneralSecurityException
+     * @throws UnsupportedEncodingException
      */
-    public synchronized SecretKey loadSecretKeyForAPI() throws IOException, GeneralSecurityException {
-        // Loading key only once for performance. If API is upgraded, it will
-        // restart the device anyway. It will load the correct key for new API.
-        if (mKey != null && mMacKey != null)
-            return mKey;
+    public String encrypt(String clearText)
+            throws GeneralSecurityException, IOException {
+        Logger.v(TAG, "Starting encryption");
 
-        final byte[] secretKeyData = AuthenticationSettings.INSTANCE.getSecretKeyData();
-        if(secretKeyData == null && Build.VERSION.SDK_INT < Build.VERSION_CODES.JELLY_BEAN_MR2) {
-            throw new IllegalStateException("Secret key must be provided for API < 18. " +
-                    "Use AuthenticationSettings.INSTANCE.setSecretKey()");
+        if (StringExtensions.IsNullOrBlank(clearText)) {
+            throw new IllegalArgumentException("Input is empty or null");
         }
 
-        if (secretKeyData != null) {
-            Logger.v(TAG, "Encryption will use secret key from Settings");
-            mKey = getSecretKey(secretKeyData);
-            mMacKey = getMacKey(mKey);
-            mBlobVersion = VERSION_USER_DEFINED;
-        } else {
-            try {
-                // androidKeyStore can store app specific self signed cert.
-                // Asymmetric cryptography is used to protect the session
-                // key for Encryption and HMac.
-                // If user specifies secret key, it will use the provided
-                // key.
-                mKey = getSecretKeyFromAndroidKeyStore();
-                mMacKey = getMacKey(mKey);
-                mBlobVersion = VERSION_ANDROID_KEY_STORE;
-            } catch (IOException | GeneralSecurityException e) {
-                Logger.e(TAG, "Failed to get private key from AndroidKeyStore", "",
-                        ADALError.ANDROIDKEYSTORE_FAILED, e);
-                throw e;
-            }
-        }
+        // load key for encryption if not loaded
+        loadSecretKeyForEncryption();
```

Comment:
If this function returns keys - you shouldn't use `mKey`. You should use what function returns